test/exe/Main.hs

```diff
@@ -229,7 +229,7 @@ diagnosticTests = testGroup "diagnostics"
       let change = TextDocumentContentChangeEvent
             { _range = Just (Range (Position 0 0) (Position 0 20))
             , _rangeLength = Nothing
-            , _text = ""
+            , _text = "module ModuleC where"
```

Comment:
Why the change here?

Reply:
these tests started failing after the changes, with the error `The IO action \8216main\8217 is not defined in module \8216Main\8217"`. Not sure why

Reply:
The one change I see compared to `hie-bios` is the linker flags in https://hackage.haskell.org/package/hie-bios-0.4.0/docs/src/HIE.Bios.Environment.html#setLinkerOptions which looks like it could be the culprit here. Did you explicitly leave them out?

Reply:
Doh, I must have missed them. Well spotted Moritz, thank you for your careful review. Breaking down the interfaces branch into smaller PRs is really paying off.

Reply:
Thanks a lot for breaking things off into smaller PRs! I know it takes a lot of effort but it makes reviews way easier.